library/psa_crypto_slot_management.h

```diff
@@ -27,7 +27,9 @@
 
 /* Number of key slots (plus one because 0 is not used).
  * The value is a compile-time constant for now, for simplicity. */
+#if !defined(PSA_KEY_SLOT_COUNT)
 #define PSA_KEY_SLOT_COUNT 32
+#endif
```

Comment:
One issue with changing this definition to be tweakable is that this definition is quite well hidden inside of an internal header file. I worry that it could make it rather confusing for users to easily find and modify these values if they are all hidden in different files.

Perhaps one solution could be to collate all of these types of different options inside one of the public config files such as is done in the default `include/mbedtls/config.h` (Line: 3535-...):
```C
/* \name SECTION: Module configuration options
 *
 * This section allows for the setting of module specific sizes and
 * configuration options. The default values are already present in the
 * relevant header files and should suffice for the regular use cases.
 *
 * Our advice is to enable options and change their values here
 * only if you have a good reason and know the consequences.
 *
 * Please check the respective header file for documentation on these
 * parameters (to prevent duplicate documentation).
 * \{
 */
/* MPI / BIGNUM options */
//#define MBEDTLS_MPI_WINDOW_SIZE            6 /**< Maximum window size used. */
//#define MBEDTLS_MPI_MAX_SIZE            1024 /**< Maximum number of bytes for usable MPIs. */
...
```
I'm sure that there are other similar tweakable options in PSA that are hidden and could be collated into a PSA config with a similar section but whether this is something we want to do this now or at a later date is another matter.

I've spoken to @ronald-cron-arm already and he seems to think similarly. Can I get your opinion here as well @gilles-peskine-arm?

Reply:
Indeed this option should be in `config.h`, with `#ifndef #define default value` in a non-user-editable header.

Another candidate for this transformation is `PSA_ITS_STORAGE_PREFIX`, which was done in a hurry but likewise should be mentioned in `config.h`.

Ideally a mention in `config.h` should require a corresponding test, but most of the existing tunable values aren't tested. So a test would be appreciated but the lack of test on CI is not a blocker.